econml/_ortho_learner.py

```diff
@@ -0,0 +1,584 @@
+    kwargs : a sequence of key-value args, with values being (numpy matrices or None)
+        Each keyword argument is of the form Var=x, with x a numpy array. Each
+        of these arrays are data variables. The model fit and predict will be
+        called with signature: `model.fit(*args, **kwargs)` and
+        `model.predict(*args, **kwargs)`. Key-value arguments that have value
+        None, are ommitted from the two calls. So all the args and the non None
+        kwargs variables must be part of the models signature.
+
+    Returns
+    -------
+    nuisances : tuple of numpy matrices
+        Each entry in the tuple is a nuisance parameter matrix. Each row i-th in the
+        matric corresponds to the value of the nuisancee parameter for the i-th input
+        sample.
+    model_list : list of objects of same type as input model
+        The cloned and fitted models for each fold. Can be used for inspection of the
+        variability of the fitted models across folds.
+    fitted_inds : np array1d
+        The indices of the arrays for which the nuisance value was calculated. This
+        corresponds to the union of the indices of the test part of each fold in
+        the input fold list.
+
+    Examples
+    --------
+
+    .. highlight:: python
+    .. code-block:: python
+
+        import numpy as np
+        from sklearn.model_selection import KFold
+        from sklearn.linear_model import Lasso
+        class Wrapper:
+            def __init__(self, model):
+                self._model = model
+            def fit(self, X, y, W=None):
+                self._model.fit(X, y)
+                return self
+            def predict(self, X, y, W=None):
+                return self._model.predict(X)
+        np.random.seed(123)
+        X = np.random.normal(size=(5000, 3))
+        y = X[:, 0] + np.random.normal(size=(5000,))
+        folds = list(KFold(2).split(X, y))
+        model = Lasso(alpha=0.01)
+        nuisance, model_list, fitted_inds = _crossfit(Wrapper(model), folds, X, y, W=y, Z=None)
+
+    >>> nuisance
+    (array([-1.1057289 , -1.53756637, -2.4518278 , ...,  1.10628792,
+       -1.82966233, -1.78227335]),)
+    >>> model_list
+    [<__main__.Wrapper object at 0x12f41e518>, <__main__.Wrapper object at 0x12f41e6d8>]
+    >>> fitted_inds
+    array([   0,    1,    2, ..., 4997, 4998, 4999])
+
+    """
+    model_list = []
+    fitted_inds = []
+    for idx, (train_idxs, test_idxs) in enumerate(folds):
+        model_list.append(clone(model, safe=False))
+        fitted_inds = np.concatenate((fitted_inds, test_idxs))
+
+        args_train = ()
+        args_test = ()
+        for var in args:
+            args_train += (var[train_idxs],) if var is not None else (None,)
+            args_test += (var[test_idxs],) if var is not None else (None,)
+
+        kwargs_train = {}
+        kwargs_test = {}
+        for key, var in kwargs.items():
+            if var is not None:
```

Comment:
I want the coders that use the _OrthoLearner to not have to have all variables in the signature of their models. For instance the _RLearner can ommit the Z. This wont cause any user error. If the user has passed in a Z=None, in the fit method and the method does not accept a Z, then it will throw an error. Also _crossfit is only called within the _OrthoLearner and there I am making sure that all variables except Y, T are called as keyword. This has to happen. Crossfit should only be called with all variables except Y, T as keywork variables. Also, every method that implements the nuisance and final models should have a signature were all variables except Y, T are keyword. I believe I made that explicit in the _OrthoLearner docstring. So the docstring you metnion of fit(X, Y, W) is not correct. It has to be fit(X=X, Y=Y, W=W). This is the contract. Shall I make it more explicit somewhere?